src/main/java/me/oyurimatheus/nossoservicodepagamento/payment/purchase/OnlinePaymentAttemptsOrder.java

```diff
@@ -0,0 +1,46 @@
+package me.oyurimatheus.nossoservicodepagamento.payment.purchase;
+
+import me.oyurimatheus.nossoservicodepagamento.payment.purchase.gateways.PaymentAttempt;
+import me.oyurimatheus.nossoservicodepagamento.payment.purchase.gateways.PaymentGatewayClient;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.stream.Collectors.toList;
+
+class OnlinePaymentAttemptsOrder {
+
+    private List<PaymentAttempt> paymentAttemptOrder;
+    private PaymentGatewayClient client;
+
+    private OnlinePaymentAttemptsOrder(List<PaymentAttempt> paymentAttemptOrder, PaymentGatewayClient client) {
+        this.paymentAttemptOrder = paymentAttemptOrder;
+        this.client = client;
+    }
+
+    public static OnlinePaymentAttemptsOrder makeGatewaysAttemptsOrderTo(PaymentGatewayClient client,
+                                                                         List<PaymentAttempt> attempts) {
+
+
+
+        List<PaymentAttempt> gatewayOrder = attempts.stream()
+                                                    .filter(PaymentAttempt::accept)
+                                                    .sorted(Comparator.comparing(PaymentAttempt::cost))
+                                                    .collect(toList());
+
+        return new OnlinePaymentAttemptsOrder(gatewayOrder, client);
+    }
+
+    public PaymentTransaction tryToPay() {
+        for (PaymentAttempt attempt: paymentAttemptOrder) {
+            Optional<PaymentTransaction> possibleTransaction = attempt.pay(client);
+            if (possibleTransaction.isPresent()) {
+                return possibleTransaction.get();
+            }
+        }
+
+        PaymentAttempt attempt = paymentAttemptOrder.get(0);
```

Comment:
Não gostei dessa linha, mas também não consegui pensar em outra forma de fazer...

Reply:
Acho que eu nem entendi essa linha... Já tem um for ali em cima tentando pagar. Como que você sabe que o que falhou foi apenas o primeiro? Pq se chegou aqui embaixo quer dizer que todos falharam...

Reply:
Você poderia registrar cada tentativa que falhou.

Reply:
É que aqui, na verdade, eu estou interessado só no pagamento. Como todos são iguais, não importa a tentativa, pensei em fazer isso para pegar o pagamento